walkthroughSteps/code-along-mentor-notes.md

```diff
@@ -223,7 +223,61 @@ Let's first write a file that gets our information from the database.
           response.end(dynamicData);
         });
     ```
-    
+
     - `getData` is asynchronous, so `response.end` should be inside it, so it doesn't run before the data comes back from the database request (same as an API request).
 
 7. Navigate to `http://localhost:3000/dynamic` to check it's worked.
+
+
+- Now, let's write another function that inserts data into the database.
+
+
+1. Inside `src/dynamic.js`, write an asynchronous function `insertData` that takes a superhero information and a callback, and returns the callback function:
+```js
+const insertData = (superHeroInfo, cb) => {
+  const { name, superPower, weight } = superHeroInfo;
+  dbConnection.query(`INSERT INTO superheroes (name, superPower, weight) VALUES (${name}, ${superPower}, ${weight})`, (err, res) => {
+    if (err) return cb(err)
+    return cb(null, res)
+  });
+};
+```
+
+2. Export `insertData` along with the previous function `getData` we wrote:
+```js
+module.exports = { getData, insertData };
+```
+
+3. Into `src/handler.js`, import function `insertData` along with `getData` from `src/dynamic.js`:
+```js
+const { getData, insertData} = require('./dynamic');
+```
+
+4. Into a handler that recieves a post request use `insertData` to capture data sent from user and store into the database:
+```js
+// supposing a handler `postSuperHero` will handle this.
+const superHeroInfo = req.body
+insertData(superHeroInfo, (err, res) => {
+  if (err) return console.log(err);
+  const insertResult = JSON.stringify(res);
+  response.writeHead(200, { 'content-type': 'application/json' });
+  response.end(insertResult);
+});
+```
+
+5. Navigate `http://localhost:3000/post-superhero` to Check your insert result.
+
+6. Security Vulnerabilities:
+
+Inserting into database as we did is a security issue, and to prevent SQL Injections, we can replace the previous query we used with this:
+```js
+// at function insertData(superHeroInfo, cb) ::
+const secured = {
+  text: `INSERT INTO superheroes (name, superPower, weight) VALUES ($1, $2, $3)`,
+  values: [name, superPower, weight]
+};
+dbConnection.query(secured, (err, res) => {
```

Comment:
does passing this `secured` object work?

I would usually do `dbConnection.query(text, values, (err, res) => { })`

as per the docs: https://node-postgres.com/features/queries#parameterized-query

Reply:
Yeah you can pass a [query config object](https://node-postgres.com/features/queries#query-config-object) with `text` and `value` properties. That might be confusing to students who have only ever seen values passed a the second argument though